dags/tpu_obs/node_pool_status.py

```diff
@@ -0,0 +1,174 @@
+
+with models.DAG(
+    dag_id="gke_node_pool_status_2",
+    start_date=datetime.datetime(2025, 7, 30),
+    schedule=constants.Schedule.WEEKDAY_PST_6PM_EXCEPT_THURSDAY,
+    catchup=False,
+    tags=["gke", "tpu-observability", "node-pool-status"],
+    description=(
+        "This DAG tests whether the status of a GKE node pool changes as"
+        " expected according to its lifecycle.It creates a node pool, waits for"
+        " it to be running, deletes a random node to trigger reconciliation,"
+        " waits for it to become running again, and finally cleans up.It also"
+        " tests the error state by creating a node pool with invalid parameters"
+        " and verifies thatthe status changes to ERROR."
+    ),
+    doc_md="""
+    ### GKE Node Pool Status Management DAG
+
+    This DAG automates the lifecycle of a GKE node pool for testing purposes.
+    This test requires an existing cluster, and if there are any other limitations
+    It will create a node-pool under the specified cluster, and clean it up after the tests
+
+    It performs the following steps:
+    1.  **Normal Path**: Creates a node pool, waits for it to be running, deletes a random node to trigger reconciliation, waits for it to become running again, and finally cleans up.
+    2.  **Error Path**: Concurrently, it attempts to create a node pool with invalid parameters to test the error state, and then cleans up.
+    """,
+) as dag:
+
+  node_pool_info = node_pool.Info(
+      project_id=models.Variable.get(
+          "PROJECT_ID", default_var=Project.TPU_PROD_ENV_ONE_VM.value
+      ),
+      cluster_name=models.Variable.get(
+          "CLUSTER_NAME", default_var="yuna-xpk-v6e-2"
+      ),
+      node_pool_name=models.Variable.get(
+          "NODE_POOL_NAME", default_var="yuna-v6e-autotest"
+      ),
+      location=models.Variable.get("LOCATION", default_var="asia-northeast1"),
+      node_locations=models.Variable.get(
+          "NODE_LOCATIONS", default_var=Zone.ASIA_NORTHEAST1_B.value
+      ),
+      num_nodes=models.Variable.get("NUM_NODES", default_var=4),
+      machine_type=models.Variable.get(
+          "MACHINE_TYPE", default_var="ct6e-standard-4t"
+      ),
+      tpu_topology=models.Variable.get("TPU_TOPOLOGY", default_var="4x4"),
+  )
+
+  problematic_node_pool_info = node_pool.Info(
+      project_id=node_pool_info.project_id,
+      cluster_name=node_pool_info.cluster_name,
+      node_pool_name=node_pool_info.node_pool_name + "-wrong",
+      location=node_pool_info.location,
+      # Choosing a region that is different from the cluster location but still
+      # compatible with the specified TPU cause the cluster creation to fail
+      # due to mismatched node locations.
+      node_locations=models.Variable.get(
+          "WRONG_NODE_LOCATION", default_var="asia-east1-c"
+      ),
+      num_nodes=node_pool_info.num_nodes,
+      machine_type=node_pool_info.machine_type,
+      tpu_topology=node_pool_info.tpu_topology,
+  )
```

Comment:
not sure if this work:
```suggestion
  problematic_node_pool_info = copy.deepcopy(node_pool_info)
  problematic_node_pool_info.node_pool_name += "-wrong"
  # <comment>
  problematic_node_pool_info.node_locations = models.Variable.get(
    "WRONG_NODE_LOCATION", default_var="asia-east1-c"
  )
```